src/App.tsx

```diff
@@ -1,35 +1,156 @@
-import { ProfilePage } from "./pages/ProfilePage/index";
+import { MyCollectionPage } from "./pages/MyCollectionPage/index";
 import { MainPage } from "./pages/MainPage/index";
+import { SignUpPage } from "./pages/SignUpPage";
+import { LoginPage } from "./pages/LoginPage";
+import { useAuth } from "./AuthContextProvider";
 
 enum Page {
   home = "Home",
-  profile = "Profile",
+  myCollection = "My Collection",
 }
 
-const allPages = Object.values(Page);
+const publicPages = [Page.home, Page.myCollection];
 
 function App() {
-  const [currentPage, setCurrentPage] = useState<Page>(Page.home);
+  const [currentPage, setCurrentPage] = useState<Page | string>(Page.home);
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const { isAuthenticated, logout } = useAuth();
+
+  const handlePageChange = (page: Page | string) => {
+    setCurrentPage(page);
+    setIsMenuOpen(false);
+  };
+
+  const handleLogout = () => {
+    logout();
+    setCurrentPage(Page.home);
+  };
 
   return (
-    <div className="class text-center font-sans bg-green-700 min-h-screen">
-      <nav className="bg-green-50 shadow-md ">
-        <ul className="flex justify-end space-x-6 p-4 xl:max-w-5xl">
-          {allPages.map((page) => (
-            <li>
+    <div className="min-h-screen bg-white text-gray-800 font-sans">
+      <nav className="bg-white shadow-md py-4 px-4 flex justify-between items-center sm:px-6">
+        <div className="flex items-center justify-between w-full sm:w-auto">
+          <span className="text-lg sm:text-xl font-bold cursor-pointer" onClick={() => handlePageChange(Page.home)}>Quote Fetcher</span>
+          <button
+            className="sm:hidden text-gray-700 focus:outline-none focus:text-blue-600"
+            onClick={() => setIsMenuOpen(!isMenuOpen)}
+            aria-label="Toggle navigation"
+          >
+            <svg
+              className="w-6 h-6"
+              fill="none"
+              stroke="currentColor"
+              viewBox="0 0 24 24"
+              xmlns="http://www.w3.org/2000/svg"
+            >
+              {isMenuOpen ? (
+                <path
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  strokeWidth="2"
+                  d="M6 18L18 6M6 6l12 12"
+                ></path>
+              ) : (
+                <path
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  strokeWidth="2"
+                  d="M4 6h16M4 12h16M4 18h16"
+                ></path>
+              )}
+            </svg>
+          </button>
+        </div>
+
+        {/* Backdrop for mobile menu */}
+        {isMenuOpen && (
+          <div
+            className="fixed inset-0 bg-black bg-opacity-50 z-30 sm:hidden"
+            onClick={() => setIsMenuOpen(false)}
+          ></div>
+        )}
+
+        {/* Mobile menu / Desktop navigation */}
+        <ul
+          className={`
+            // Mobile styles
```

Comment:
It's really hard to understand the code in the component due to high amount of styles. Would be better to create a dedicated components for navbar elements - then logic would be in the App component but styles would be primarily hidden in these dedicated UI components.